packages/flutter_tools/lib/src/isolated/native_assets/targets.dart

```diff
@@ -0,0 +1,378 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart' show FileSystem;
+import 'package:native_assets_cli/code_assets_builder.dart'
+    show
+        AndroidCodeConfig,
+        Architecture,
+        CCompilerConfig,
+        CodeAssetExtension,
+        IOSCodeConfig,
+        LinkModePreference,
+        MacOSCodeConfig,
+        OS,
+        ProtocolExtension;
+import 'package:native_assets_cli/data_assets_builder.dart'
+    show DataAssetsExtension, ProtocolExtension;
+
+import '../../base/common.dart' show throwToolExit;
+import '../../build_info.dart'
+    show
+        AndroidArch,
+        DarwinArch,
+        EnvironmentType,
+        TargetPlatform,
+        getAndroidArchForName,
+        getDarwinArchsFromEnv,
+        getIOSArchForName,
+        kAndroidArchs,
+        kIosArchs,
+        kSdkRoot;
+import '../../build_system/exceptions.dart' show MissingDefineException;
+import '../../macos/xcode.dart' as xcode show environmentTypeFromSdkroot;
+import 'android/native_assets.dart'
+    show cCompilerConfigAndroid, getNativeAndroidArchitecture, targetAndroidNdkApi;
+import 'ios/native_assets.dart' show getIOSSdk, getNativeIOSArchitecture, targetIOSVersion;
+import 'linux/native_assets.dart';
+import 'macos/native_assets.dart' show getNativeMacOSArchitecture, targetMacOSVersion;
+import 'macos/native_assets_host.dart';
+import 'native_assets.dart';
+import 'windows/native_assets.dart';
+
+/// This is a translation layer between Flutter, which knows only
+/// [TargetPlatform]s, and `dart-lang/native`, which knows only asset types and
+/// how to build them based on things like [OS]s and [Architecture]s.
+sealed class AssetBuildTarget {
+  const AssetBuildTarget({required this.supportedAssetTypes});
+
+  final List<SupportedAssetTypes> supportedAssetTypes;
+
+  List<ProtocolExtension> get extensions;
+
+  String get targetString;
+
+  List<DataAssetsExtension> get dataAssetExtensions => <DataAssetsExtension>[
+    if (supportedAssetTypes.contains(SupportedAssetTypes.dataAssets)) DataAssetsExtension(),
+  ];
+
+  /// Build the list of [AssetBuildTarget]s for a given [TargetPlatform].
+  ///
+  /// It needs access to other parameters such as the [fileSystem] or
+  /// [environmentDefines] to retrieve options for some of the targets.
+  static List<AssetBuildTarget> targetsFor(
+    TargetPlatform targetPlatform,
```

Comment:
Nit: consider using required named parameters for functions that take more than a single, obvious parameter as input.